src/Craned/Core/JobManager.cpp

```
@@ -47,6 +47,192 @@ constexpr uint64_t kPendingSigchldMask = 1ULL << (SIGCHLD - 1);
+  if (!JobHasLocalCpuPool(*job)) return;
+  if (job->job_local_overflow_initialized) return;
+  job->job_local_overflow_cores = GetJobCoreIds(*job);
+  job->job_local_overflow_initialized = true;
+}
+
+void ClaimJobLocalCores(JobInD* job, const std::set<uint32_t>& core_ids) {
+  InitJobLocalOverflow(job);
+  for (auto cpu_id : core_ids) job->job_local_overflow_cores.erase(cpu_id);
+}
+
+void ReleaseJobLocalCores(JobInD* job, const std::set<uint32_t>& core_ids) {
```

Cppcheck (2.21.0)

[error] Found an exit path from function with non-void return type that has missing return statement

(missingReturn)